SharpPcap/LibPcap/PcapDeviceCaptureLoop.cs

```diff
@@ -123,13 +123,37 @@ public void Capture(int packetCount)
             m_pcapPacketCount = Pcap.InfinitePacketCount;
         }
 
-        internal protected virtual bool UsePosixPoll
+        /// <summary>
+        /// unix specific code, we want to poll for packets
+        /// otherwise if we call pcap_dispatch() the read() will block
+        /// and won't resume until a packet arrives OR until a signal
+        /// occurs
+        /// </summary>
+        /// <param name="timeout">
+        /// Timeout chosen to allow the capture thread to loop frequently enough
+        /// to enable it to properly exit when the user requests it to but
+        /// infrequently enough to cause any noticable performance overhead
+        /// </param>
+        /// <returns>true if poll was successfull and we have data to read, false otherwise</returns>
+        protected internal bool PollFileDescriptor(int timeout = 500)
```

Comment:
we can probably fairly make this 100 or even 50ms by default if you wanted to.

Reply:
This is intentionally 500ms, half of `const int readTimeoutMilliseconds = 1000;`.
Making it too small like 50ms would cause too many system calls.